nexus/src/db/datastore.rs

```diff
@@ -1607,19 +1625,24 @@ impl DataStore {
                 insert_new_query.insert_and_get_results(conn).map_err(|e| {
                     match e {
                         SyncInsertError::CollectionNotFound => {
-                            diesel::result::Error::RollbackTransaction
```

Comment:
I'm not certain the usage of this error was correct.

From https://docs.diesel.rs/master/diesel/result/enum.Error.html#variant.RollbackError:
> An error occurred during the rollback of a transaction.
> An example of when this error would be returned is if a rollback has already be called on the current transaction.

"Not found" seems like a much more appropriate error to propagate from the transaction. Although this is a relatively simple case, there definitely could be a wider set of returnable errors than just those within `diesel::result::Error`.

Reply:
I think you're looking at RollbackError, which is different from RollbackTransaction

Reply:
Ah, whoops. Yeah, you're right. `RollbackTransaction` definitely is an okay error to return here, there just might be situations where we want to report something more through the error.

Reply:
Yeah, I think returning a more specific thing is definitely fine. I think at the time I didn't dig into whether returning other errors rollsback the transaction (since I think it's not in the `transaction()` documentation), but it definitely does work that way so we should do better